test/trafos/backends/pywt_bindings_test.py

```diff
@@ -218,5 +220,43 @@ def test_multilevel_decomp_inverts_recon(shape_setup):
     assert all_almost_equal(coeffs, wave_decomp)
 
 
+def test_explicit_example():
+    """Exhaustive test of a hand-calculated 2D example."""
+
+    x = np.array([[1, 1, 0],
+                  [1, 0, 1],
+                  [0, 1, 1]])
+
+    # Explicit Haar wavelet filters
+    tau = 1.0 / np.sqrt(2)
+    filter_l = np.array([tau, tau])
+    filter_h = np.array([-tau, tau])
+
+    # Build the 2D filters
+    filter_ll = filter_l[:, None] * filter_l[None, :]
+    filter_lh = filter_l[:, None] * filter_h[None, :]
+    filter_hl = filter_h[:, None] * filter_l[None, :]
+    filter_hh = filter_h[:, None] * filter_h[None, :]
+
+    # Convolve x with 2D filters (implicitly uses zero-padding)
+    conv_ll = convolve(x, filter_ll)
+    conv_lh = convolve(x, filter_lh)
+    conv_hl = convolve(x, filter_hl)
+    conv_hh = convolve(x, filter_hh)
+
+    # Downsampling gives the wavelet coefficients (starting with index 1)
```

Comment:
That's the choice that PyWavelets took. I think it's rather arbitrary, but I may be wrong and it's prescribed by the theory for discrete wavelet transforms.